public/app/features/explore/state/datasource.ts

```diff
@@ -41,7 +41,7 @@ export function changeDatasource(
   return async (dispatch, getState) => {
     const orgId = getState().user.orgId;
     const { history, instance } = await loadAndInitDatasource(orgId, datasourceName);
-    const currentDataSourceInstance = getState().explore[exploreId].datasourceInstance;
+    const currentDataSourceInstance = getState().explore[exploreId]!.datasourceInstance;
```

Comment:
So something like
```
function getDataSourceInstance(state, exploreId) {
  return state.explore[exploreId].datasourceInstance;
}

getDataSourceInstance(getState(), exploreId)
```
? 

I mean I don't know, feel like it doesn't really help much. I mean there is no need to cache anything here so it would be just hiding the accessors while the usage still needs to list all the words except for `explore` specifically.

Reply:
I personally don't mind `getState().explore[exploreId]!...`.

Reply:
It was just a nit, I personally try to use selectors as much as possible, but here there's really no benefit in doing it 🙂